src/NumberGuessingGame.js

```diff
@@ -1,77 +1,76 @@
-  constructor(props) {
-    super(props);
+const getRandomNumber = Math.floor(Math.random() * 100) + 1
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
+export default function NumberGuessingGame  () {
+  const [currentGuess, setCurrentGuess] = useState("");
+  const [latestGuess, setLatestGuess] = useState("");
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0);
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber);
+  const [isCorrectGuess, setIsCorrectGuess] = useState(false);
+  const [isGameOver, setIsGameOver] = useState(false);
+  const [showHint, setShowHint] = useState(false);
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
 
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
+  const handleGuess = (e) => {
+    e.preventDefault();
+    if (Number(currentGuess) === numberToGuess) {
+      setIsGameOver(true);
+      setIsCorrectGuess(true);
+    }
+    else {
+      setShowHint(true);
+      setLatestGuess(currentGuess);
+      if (numberOfGuesses === MAX_ATTEMPTS) {
+        setIsGameOver(true);
+      }
+    }
+    setNumberOfGuesses(numberOfGuesses + 1);
   }
 
-  handleReset() {
-    this.setState({
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    });
+  const onReset = () => {
+    setNumberToGuess(getRandomNumber);
+    setCurrentGuess("");
+    setNumberOfGuesses(0);
+    setIsCorrectGuess(false);
+    setIsGameOver(false);
+    setShowHint(false);
   }
-
-  render() {
-    const isCorrectGuess = this.state.latestGuess === this.state.numberToGuess;
-
-    const isGameOver =
-      isCorrectGuess || this.state.numberOfGuesses === MAX_ATTEMPTS;
-
     return (
       <div>
         <h2>I'm thinking of a number from 1 to 100.</h2>
         <h2>
-          Can you guess the number I am thinking of in {MAX_ATTEMPTS} tries?
+          Can you guess the number I am thinking of in {MAX_ATTEMPTS+1} tries?
         </h2>
-        <GuessControl onGuess={this.handleGuess} />
-        {isGameOver && (
-          <GameOver hasWon={isCorrectGuess} onReset={this.handleReset} />
-        )}
-        {!isGameOver && (
-          <GuessMessage
-            guess={this.state.latestGuess}
-            numberToGuess={this.state.numberToGuess}
-            numberOfGuesses={this.state.numberOfGuesses}
-          />
-        )}
+        <p> Guesses so far: {numberOfGuesses}</p>
+        <GuessControl 
+        onGuess={handleGuess} 
+        setCurrentGuess={setCurrentGuess} 
+        currentGuess={currentGuess} 
+        setShowHint={setShowHint}
+        isGameOver={isGameOver}
+        />
```

Comment:
You might want to hide this when the game is over or if the user wins the game 👍